instrumentation/aws-sdk/aws-sdk-1.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/awssdk/v1_11/AwsHttpClientInstrumentation.java

```diff
@@ -38,22 +40,26 @@
   @Override
   public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
     return singletonMap(
-        isMethod().and(not(isAbstract())).and(named("doExecute")),
+        isMethod()
+            .and(not(isAbstract()))
+            .and(named("doExecute"))
+            .and(takesArgument(0, named("com.amazonaws.Request")))
+            .and(returns(named("com.amazonaws.Response"))),
         AwsHttpClientInstrumentation.class.getName() + "$HttpClientAdvice");
   }
 
   public static class HttpClientAdvice {
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void methodExit(
-        @Advice.Argument(value = 0, optional = true) Request<?> request,
+        @Advice.Argument(value = 0) Request<?> request,
+        @Advice.Return Response<?> response,
         @Advice.Thrown Throwable throwable) {
-      if (throwable != null) {
-        ContextScopePair scope = request.getHandlerContext(CONTEXT_SCOPE_PAIR_CONTEXT_KEY);
-        if (scope != null) {
-          request.addHandlerContext(CONTEXT_SCOPE_PAIR_CONTEXT_KEY, null);
-          tracer().endExceptionally(scope.getContext(), throwable);
-          scope.closeScope();
-        }
+      if (throwable instanceof Exception) {
```

Comment:
There is a small regression in my desire to delegate the actual tracing to the library instrumentation - `Error` will result in unclosed traces. I feel this should be rare enough to live with until we get a report of problems.